pkg/skaffold/deploy/resource/deployment.go

```diff
@@ -74,33 +57,24 @@ func (d *Deployment) UpdateStatus(details string, err error) {
 	}
 }
 
-func (d *Deployment) IsStatusCheckComplete() bool {
-	return d.done
-}
-
-func (d *Deployment) ReportSinceLastUpdated() string {
-	if d.status.reported {
-		return ""
+func NewDeployment(name string, ns string, deadline time.Duration) *Deployment {
+	return &Deployment{
+		Base: &Base{
```

Comment:
might be worth it to create a constructor for this

```go
func NewBase(name, namespace ...`
```

in case you end up adding stuff to `Base` later on it might be easier (had that experience with `portForwardEntry`)